packages/ddex/src/controllers/uploadsController.ts

```diff
@@ -0,0 +1,139 @@
+          const zipFileUUID = uuidv4()
+          for (const file of files) {
+            if (file.path.endsWith('.xml')) {
+              xmlProcessorService.addXmlFile(file.data, uploadedBy, zipFileUUID)
+            }
+          }
+        } else {
+          return res.status(400).json({ error: 'Unsupported file type.' })
+        }
+
+        res.json({ message: 'File uploaded and processed successfully.' })
+      } catch (error: any) {
+        return res.status(500).json({ error: error.message })
+      }
+    })
+  }
+}
+
+export const getUploads = (sql: Sql) => async (req: Request, res: Response) => {
+  const { status, nextId, prevId, limit = '10' } = req.query
+
+  const numericLimit = Number(limit)
+  const numericNextId = Number(nextId || 0)
+  const numericPrevId = Number(prevId || 0)
+
+  if ((nextId && isNaN(numericNextId)) || (prevId && isNaN(numericPrevId))) {
+    return res.status(400).json({ error: 'Invalid pagination parameters.' })
+  }
+
+  if (!Number.isInteger(numericLimit)) {
+    return res.status(400).json({ error: 'Invalid limit.' })
+  }
+
+  try {
+    const uploads = await sql<XmlFileRow[]>`
+        SELECT * FROM xml_files
+        ${
+          status &&
+          (status === 'success' || status === 'pending' || status === 'error')
+            ? sql` WHERE status = ${status}`
+            : sql``
+        }
+        ${
+          numericNextId
+            ? sql` AND id < ${numericNextId} ORDER BY id DESC`
+            : sql``
+        }
+        ${
+          numericPrevId
+            ? sql` AND id > ${numericPrevId} ORDER BY id ASC`
+            : sql``
+        }
+        LIMIT ${numericLimit}
+      `
+    res.json(uploads)
+  } catch (error) {
+    console.error('Query error:', error)
+    res.status(500).json({ error: 'Internal server error' })
+  }
+}
+
+export const getReleases =
+  (sql: Sql) => async (req: Request, res: Response) => {
+    const { status, nextCursor, prevCursor, limit = 10 } = req.query
+
+    // Convert limit to a number and validate
+    const numericLimit = Number(limit)
+    if (!Number.isInteger(numericLimit)) {
+      return res.status(400).json({ error: 'Invalid limit.' })
+    }
+
+    let whereConditions = sql``
+
+    if (
+      status &&
+      (status === 'success' || status === 'processing' || status === 'error')
+    ) {
+      whereConditions = sql` WHERE status = ${status}`
+    }
+
+    if (nextCursor && typeof nextCursor === 'string') {
+      const [nextDate, nextId] = nextCursor.split(',')
+      whereConditions = sql`${whereConditions} AND (release_date, id) < (${nextDate}, ${nextId})`
+    } else if (prevCursor && typeof prevCursor === 'string') {
+      const [prevDate, prevId] = prevCursor.split(',')
+      whereConditions = sql`${whereConditions} AND (release_date, id) > (${prevDate}, ${prevId})`
+    }
+
+    const orderBy =
+      nextCursor || prevCursor
+        ? sql` ORDER BY release_date ${nextCursor ? 'DESC' : 'ASC'}, id ${
```

Comment:
you right

Reply:
ok so actually yes the end result on the page should be the same order (descending), BUT when paginating forward (checking where id > X) it needs to do ascending or else it would always select the highest elements (`select * from table where id > X desc` is the same as `select * from table desc`)

example for
```
100
99
...
2
1
```
if we're currently displaying `40,41...48,49` and want to display `50,51...58,59`, then it needs to get `where id > 49 asc` and reverse it so on the client it ends up displaying as descending still like:

```
59
58
...
51
50
```

followed up in https://github.com/AudiusProject/audius-protocol/pull/7133